### ISSUE TITLE
allow for different common guitar tunings

### ISSUE THREAD

**Issue by RustoMCSpit** [thetwom/Tuner#77]

**Comment by RustoMCSpit** [comment]
as in, when yiu select a tuning preset

**Comment by thetwom** [comment]
You can add custom tunings. Go to the overview of the different tunings and press "+" at the bottom right. Then put together the notes you need. Give it a name and an icon. 

Or did you mean something else?

**Comment by RustoMCSpit** [comment]
> You can add custom tunings. Go to the overview of the different tunings and press "+" at the bottom right. Then put together the notes you need. Give it a name and an icon.
> 
> Or did you mean something else?

oh i meant having them included by default under a drop down for guitar

**Comment by thetwom** [comment]
Thanks for the clarification.

I tend not to include tuning variants since this  quickly turns the instruments list quite long. For variants the custom tuning functionality is made. You can even directly copy the guitar tuning (by pressing the copy icon to the right) and just modify the notes needed.

**Comment by RustoMCSpit** [comment]
> Thanks for the clarification.
> 
> I tend not to include tuning variants since this quickly turns the instruments list quite long. For variants the custom tuning functionality is made. You can even directly copy the guitar tuning (by pressing the copy icon to the right) and just modify the notes needed.

thats why it should be a dropdown on the guitar tab

**Comment by thetwom** [comment]
I do not really see the advantage to custom tunings. Predefined tunings are more a list of very common tunings, even though you can already argue, what to include and what not. People, who play more advanced tunings should easily be able to create the tunings they need. Or directly go for chromatic, since they know anyway what notes they want to tune.

**Comment by Glandos** [comment]
Maybe this need can be solved by import/export of a set of tunings?

**Comment by RustoMCSpit** [comment]
> Maybe this need can be solved by import/export of a set of tunings?

a drop down of tunings on each instrument seems way more logical and easier to implement

**Comment by wolftune** [comment]
FWIW, a norm of having various tunings for an instrument such as in apps like this is something that helps break the hegemony of standard tunings. It makes it clearer that other tunings are not obscure and pointless but are a normal thing to explore.

I otherwise appreciate the value of having standards and of not throwing everything into it. It's not nice to be presented with hundreds or thousands of options for a setting

**Comment by RustoMCSpit** [comment]
this is being worked on right now.

a few things:

you should be able to drag strings up and down.
you should be able to import / export.
you should have drop downs for the instruments to show different tunings.
you should be able to add tunings to those instruments yourself.
you should be able to set a default tuning for that instrument where if you dont click on that instruments drop down you automatically get that tuning.
you should be able to hide instruments.
you should be able to show multiple tunings for the same instrument via starring it.

**Comment by RustoMCSpit** [comment]
lately detected tones as a feature could be great for automatic scale detection?

**Comment by thetwom** [comment]
> FWIW, a norm of having various tunings for an instrument such as in apps like this is something that helps break the hegemony of standard tunings. It makes it clearer that other tunings are not obscure and pointless but are a normal thing to explore.
> 
> I otherwise appreciate the value of having standards and of not throwing everything into it. It's not nice to be presented with hundreds or thousands of options for a setting

I guess this is a topic, where one could discuss endlessy. As you say it is about having standards against not always conforming to standards ... the philosophy I have been following up to now: Provide a few defaults for the absolute beginners and anyone who is more experienced can define own tunings or uses the chromatic tuner anyway. Not sure, if it is really the task of a tuner to be educational here by suggesting other tunings. This can quickly degrade the user experience (especially if you start being educational also at other places). 

> you should be able to drag strings up and down.

Agree that this can help, the difficult thing is always the UI design. Drag and drop would be an option but the support for this is not very good in the standard components, so it needs quite some extra development.  I guessed, that this is not needed very often, so I put focus on other things :-).

> you should be able to import / export.

There is an import / export option. Too difficult to find? It is in the instrument overview. Long press on a instrument to only export a selection of instruments. Or did you mean something else?

> you should have drop downs for the instruments to show different tunings.
> you should be able to add tunings to those instruments yourself.
> you should be able to set a default tuning for that instrument where if you dont click on that instruments drop down you automatically get that tuning.

Good question, if it is worth having this extra nesting. I am not sure, how much people are managing different instruments that this is really needed. I somehow can't imagine that people would like to go through an extra nesting level to find the correct instruments, but would rather use a chromatic tuner. But on the other side, there are always valid use cases, which I do not have in mind :-). 

> you should be able to hide instruments.
> you should be able to show multiple tunings for the same instrument via starring it.

This all points to much more elaborate managing of instruments. At the moment, the tuner targets only very simple cases. At this point we are at the question, where it is worth putting effort. Unfortunately, it is very difficult to get information about what features are really used. This app does not collect any usage information, so I have zero insight apart from what people report here. So keep on asking :-), but in the end I cannot promise much ...

**Comment by wolftune** [comment]
I also agree generally that anyone can just stick to chromatic. I never really understood the idea of instrument-specific tuner settings at all actually
